packages/lib/utils/chunkGenerator.ts

```diff
@@ -0,0 +1,49 @@
+import { read, open, stat } from 'fs';
+import { promisify } from 'util';
+
+const readPromisify = promisify(read);
+const statPromisify = promisify(stat);
+const openPromisify = promisify(open);
+
+
+async function* generateChunks(
+	filePath: string,
+	size: number,
+// eslint-disable-next-line no-undef
```

Comment:
This needs to be solved properly. If AsyncGenerator is undefined maybe we can't use at the moment. For example is it even supported in React Native?